G/GAP_pkg/GAP_pkg_caratinterface/build_tarballs.jl

```diff
@@ -0,0 +1,136 @@
+                  "fdbc0f86befd8bf575c93475b33f58329ce99fe2ef97ba69af1478bc8664059c"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd CaratInterface*
+
+# HACK to workaround need to pass --with-gaproot
+mkdir -p $prefix/lib/gap/ # HACK
+echo "GAParch=dummy" > $prefix/lib/gap/sysinfo.gap # HACK
+echo "GAP_CPPFLAGS=dummy" >> $prefix/lib/gap/sysinfo.gap # HACK
+
+./configure ${prefix}/lib/gap
+make -j${nproc}
+
+# copy just the executable
+mkdir -p ${prefix}/bin/
+cp bin/*/* ${prefix}/bin/
+
+install_license LICENSE
+
+rm $prefix/lib/gap/sysinfo.gap
+"""
+
+name = gap_pkg_name(name)
+
+platforms = supported_platforms()
+filter!(p -> nbits(p) == 64, platforms) # we only care about 64bit builds
+filter!(!Sys.iswindows, platforms)      # Windows is not supported
+
+dependencies = Dependency[
+]
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("Add", :Add),
+    ExecutableProduct("Aut_grp", :Aut_grp),
+    ExecutableProduct("Bravais_catalog", :Bravais_catalog),
+    ExecutableProduct("Bravais_equiv", :Bravais_equiv),
+    ExecutableProduct("Bravais_grp", :Bravais_grp),
+    ExecutableProduct("Bravais_inclusions", :Bravais_inclusions),
+    ExecutableProduct("Bravais_type", :Bravais_type),
+    ExecutableProduct("Conj", :Conj),
+    ExecutableProduct("Conj_bravais", :Conj_bravais),
+    ExecutableProduct("Conjugated", :Conjugated),
+    ExecutableProduct("Conv", :Conv),
+    ExecutableProduct("Datei", :Datei),
+    ExecutableProduct("Elt", :Elt),
+    ExecutableProduct("Extensions", :Extensions),
+    ExecutableProduct("Extract", :Extract),
+    ExecutableProduct("First_perfect", :First_perfect),
+    ExecutableProduct("Form_elt", :Form_elt),
+    ExecutableProduct("Form_space", :Form_space),
+    ExecutableProduct("Formtovec", :Formtovec),
+    ExecutableProduct("Full", :Full),
+    ExecutableProduct("Gauss", :Gauss),
+    ExecutableProduct("Graph", :Graph),
+    ExecutableProduct("Idem", :Idem),
+    ExecutableProduct("Inv", :Inv),
+    ExecutableProduct("Invar_space", :Invar_space),
+    ExecutableProduct("Is_finite", :Is_finite),
+    ExecutableProduct("Isometry", :Isometry),
+    ExecutableProduct("KSubgroups", :KSubgroups),
+    ExecutableProduct("KSupergroups", :KSupergroups),
+    ExecutableProduct("Kron", :Kron),
+    ExecutableProduct("Long_solve", :Long_solve),
+    ExecutableProduct("Ltm", :Ltm),
+    ExecutableProduct("Mink_red", :Mink_red),
+    ExecutableProduct("Minpol", :Minpol),
+    ExecutableProduct("Modp", :Modp),
+    ExecutableProduct("Mtl", :Mtl),
+    ExecutableProduct("Mul", :Mul),
+    ExecutableProduct("Name", :Name),
+    ExecutableProduct("Normalizer", :Normalizer),
+    ExecutableProduct("Normalizer_in_N", :Normalizer_in_N),
+    ExecutableProduct("Normlin", :Normlin),
+    ExecutableProduct("Orbit", :Orbit),
+    ExecutableProduct("Order", :Order),
```

Comment:
This apparently results in the error
```
ERROR: LoadError: `Order` is already defined in Base
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] ExecutableProduct(binnames::Vector{String}, varname::Symbol, dir_path::Nothing)
   @ BinaryBuilderBase /cache/julia-buildkite-plugin/depots/e2fd9734-29d8-45cd-b0eb-59f7104f3131/packages/BinaryBuilderBase/LieLT/src/Products.jl:356
 [3] ExecutableProduct(binnames::Vector{String}, varname::Symbol)
   @ BinaryBuilderBase /cache/julia-buildkite-plugin/depots/e2fd9734-29d8-45cd-b0eb-59f7104f3131/packages/BinaryBuilderBase/LieLT/src/Products.jl:355
 [4] ExecutableProduct(::String, ::Symbol)
   @ BinaryBuilderBase /cache/julia-buildkite-plugin/depots/e2fd9734-29d8-45cd-b0eb-59f7104f3131/packages/BinaryBuilderBase/LieLT/src/Products.jl:364
 [5] top-level scope
   @ /cache/build/yggy-amdci7-0/julialang/yggdrasil/G/GAP_pkg/GAP_pkg_caratinterface/build_tarballs.jl:50
in expression starting at /cache/build/yggy-amdci7-0/julialang/yggdrasil/G/GAP_pkg/GAP_pkg_caratinterface/build_tarballs.jl:50
```
and indeed we have this in BinaryBuilderBase:
```
    function ExecutableProduct(binnames::Vector{String}, varname::Symbol, dir_path::Union{AbstractString, Nothing}=nothing)
        if isdefined(Base, varname)
            error("`$(varname)` is already defined in Base")
        end
```
But why? And is this documented somewhere?

Any pointers, @giordano 

(I'll work around it here by renaming, presumably also for `Symbol` below, it is no big issue; but I still am curious as to why this is an error in the first place)

Reply:
Because you can't refer to a variable with the same name as a variable exported by Base (unless you qualify it with the module, but we also export the var), we had issues in the past. But perhaps the check can be refined to limit it to exported symbols.